File: 6. Zigzag Conversion/main.js
/**
 * @param {string} s
 * @param {number} numRows
 * @return {string}
 */
var convert = function (s, numRows) {
  if (numRows == 1) return s;
  let output = [];

  for (let i = 0; i < numRows; ++i) {
    output.push("");
  }

  let rowCounter = 0;
  let angleCounter = 0;
  for (let i = 0; i < s.length; ++i) {
    if (rowCounter <= numRows - 1) {
      output[rowCounter] += s[i];
      ++rowCounter;
    } else {
      if (numRows === 2) {
        --i;
        rowCounter = 0;
        continue;
      }
      output[numRows - angleCounter - 2] += s[i];
      ++angleCounter;

      if (angleCounter >= numRows - 2) {
        angleCounter = 0;
        rowCounter = 0;
      }
    }
  }

  let outputS = "";
  for (let i = 0; i < numRows; ++i) {
    outputS += output[i];
  }
  return outputS;
};

File: 6. Zigzag Conversion/main.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        

        if numRows == 1 or numRows >= len(s):
            return s

        strRows = []
        output = ""

        for i in range(numRows):
            strRows.append("")
        
        rowNum = 0
        flipDir = False

        for i in range(len(s)):
            strRows[rowNum] += s[i]

            if flipDir:
                rowNum -= 1
            else:
                rowNum += 1
            
            if rowNum == 0 or rowNum == (numRows - 1):
                flipDir = not flipDir

        for i in range(numRows):
            output += strRows[i]

        return output